### DISCUSSION TITLE
How to rename the model name in the client?

### DISCUSSION THREAD

**Discussion by tbhaxor** [prisma/prisma#22601] [Q&A] [answered]
### Question

I have a model in the schema.prisma

```
generator client {
    provider      = "prisma-client-js"
    output        = "generated"
    binaryTargets = ["native", "linux-musl-openssl-3.0.x", "debian-openssl-3.0.x"]
}

model APIKey {
    id      BigInt @id @default(autoincrement())
    name    String
    key     String @db.Uuid
    user_id String @db.Uuid

    created_at DateTime? @default(now()) @db.Timestamptz()
    updated_at DateTime? @updatedAt @db.Timestamptz()

    @@map("api_keys")
}
```

When I generate it and use in the client it gives different name

```js
const db = new PrismaClient();
db.aPIKey
```

### Information about Prisma Schema, Client Queries and Environment (optional)

- OS: Ubuntu
- Database: Postgres
- Node.js version: v18.18.2
- Prisma version: v5.7.1

**Comment by nurul3101** [comment]
Hello @tbhaxor 👋 

The `@@map` attribute is used to map the model to a different table name **in the database**. It doesn't change the name of the model as it appears in the Prisma Client. It only affects the underlying database table name.

There is no built-in way to rename the model in the Prisma Client API without changing the model name in the schema itself. Do you want to name the model as `APIKey` but want to access it in PrismaClient as `aPIKey`?

**Reply by tbhaxor** [reply]
I want to name the model APIKey, but it should be accessible as apiKey. I am ready to change the model name which can give me `apiKey` in the prisma client.

Also I think adding an option to rename this would be a good idea.

**Reply by tbhaxor** [reply, marked as answer]
I have resolved it using `model ApiKey`
